tests/func.sh

```diff
@@ -357,14 +357,33 @@ check() {
 		max=`cat /proc/sys/dev/raid/speed_limit_max`
 		echo 200000 > /proc/sys/dev/raid/speed_limit_max
 		sleep 0.1
-		while grep -Eq '(resync|recovery|reshape|check|repair) *=' /proc/mdstat ||
-			grep -v idle > /dev/null /sys/block/md*/md/sync_action
-		do
-			sleep 0.5
+		max_iterations=5
+		# wait up to 5 seconds for one of action appeared in sync_action,
+		# which will trigger grow-continue process
+		for ((i = 0 ; i < max_iterations ; i++ )); do
+			sync_action=`grep -Eq '(resync|recovery|reshape|check|repair) *=' /proc/mdstat || grep -v idle > /dev/null /sys/block/md*/md/sync_action`
+			if [ "$sync_action" ]; then
+				break
+			fi
+			sleep 1
 		done
+
+		if [[ i == max_iterations ]]; then
+			echo >&2 "Timeout waiting for reshape operation appeared in sync_action"
+			exit 1
+		fi
+
+		# wait for grow-continue to finish but break if sync_action does not
+		# contain any reshape value
 		while ps auxf | grep "mdadm --grow --continue" | grep -v grep
```

Comment:
We could possibly use pgrep here like
`while pgrep -f "mdadm --grow --continue" > /dev/null`
...but I'm fine with leaving this as it is now.